### PR TITLE
many: rename apply-config hook to configure

### PR DESCRIPTION
**Pull request by kyrofa**
This PR finishes work on LP: [#1596629](https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/1596629) by renaming the `apply-config` hook to `configure`. It also removes `check-config` as a valid hook, since checking will be done in `configure`.

### CODE REVIEW

File: overlord/configstate/applyconfighandler.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ package configstate
 
 import "github.com/snapcore/snapd/overlord/hookstate"
 
-// applyConfigHandler is the handler for the apply-config hook.
+// applyConfigHandler is the handler for the configure hook.
```
Comment (pedronis) [inline overlord/configstate/applyconfighandler.go]: would be could to rename this to to configureHandler etc

Comment (niemeyer) [inline overlord/configstate/applyconfighandler.go]: Tracked this one in the "Configuration support" card:

https://trello.com/c/PpM4XRUQ/30-configuration-support

**Comment by niemeyer** [comment]
Merging to unblock the SRU. Tracked the pending item in the config card.

File: overlord/configstate/applyconfighandler.go (hunk 1 above)
Comment (kyrofa) [inline overlord/configstate/applyconfighandler.go]: Done in #2023.

**Pull request merged**